### PR TITLE
Add `helm get metadata` command

### LINKED ISSUE #11625: Introduce `helm get metadata` command
### Problem
There's a command `helm get` with a number of subcommands:
* `values`
* `notes`
* `manifest`
* `hooks`
* `all`

None of them return the chart version that was used. The chart version is returned in `helm list` though.
None of them return the namespace or revision or helm chart app version as well.
Deploy timestamp is returned, but only by `all` command, which does not have `--output` flag, so it can't output to machine-readable JSON or YAML formats.

### Proposal
Introduce `helm get metadata RELEASE_NAME [flags]` command:
* to return the following release attributes:
  * `name` -- the release name
  * `chart` -- the chart name (note: just a chart name; chart version goes in a separate attribute)
  * `version` -- the helm chart `version` field
  * `appVersion` -- the helm chart `appVersion` field
  * `namespace` -- the namespace where the chart is installed into
  * `revision` -- the revision of the release
  * `status` -- the release status
* to have `--output` flag with options:
  * `table`  -- default
  * `json`
  * `yaml`

### PR DESCRIPTION
**Pull request by mih-kopylov**
Closes #11625

**What this PR does / why we need it**:
The PR introduces a new command `helm get metadata`

**Special notes for your reviewer**:

**If applicable**:
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by technosophos — APPROVED** [review]
Code review looks good, and this is a useful feature to have. If @mattfarina is also good with this, I think we can include it. (Do we need any changes to the description for `all` to note that this data is not included with `all`?)

File: cmd/helm/get_metadata.go
Comment (sabre1041) [inline cmd/helm/get_metadata.go:80]: Since `appVersion` is optional. it will present a blank space here. Is there a desire to potentially omit empty fields if they are not populated?

File: cmd/helm/get.go
Comment (sabre1041) [inline cmd/helm/get.go:51]: Shouldn't the `getHelp` variable in this file also be updated to account for new functionality being added?

**Review by sabre1041 — COMMENTED** [review]
This is a good start. However, there appears to be pieces missing to incorporate into the larger lifecycle of the `helm get` subcommand.

Shouldn't properties included in the `helm get metadata` command now being exposed also appear in `helm get all`?

File: cmd/helm/get_metadata.go
Comment (mih-kopylov) [reply cmd/helm/get_metadata.go:80]: IMHO, it's better to print an empty string so that it's explicitly stated that the filed is here and its value is empty, however, I'm ok to change the implementation to omit the empty field.
Up to you to decide.

File: cmd/helm/get.go
Comment (mih-kopylov) [reply cmd/helm/get.go:52]: Thanks, done

**Comment by mih-kopylov** [comment]
@sabre1041 

> Shouldn't properties included in the helm get metadata command now being exposed also appear in helm get all?

Yeah, makes sense. I've added missed properties to `get all` command output, however the fields names there differ from the `get metadata` output, and I didn't have an intention to make them same because:
* LAST DEPLOYED is printed with a space in `get all` command, that's not common to have spaces in keys when printed in json
* time is printed in not a standard format, while I used time.RFC3339 format intentionally in order to keep it machine-readable

Please let me know if you're ok with that

**Comment by sabre1041** [comment]
@mih-kopylov Still on the fence regarding `APP_VERSION`. However, since more than not, that field will be populated, lets move this one forward.

Thanks for contributing this enhancement.

**Review by sabre1041 — APPROVED** [review]

**Comment by mih-kopylov** [comment]
@technosophos any change to include this PR to the upcoming helm:3.11.0?

**Comment by mih-kopylov** [comment]
@sabre1041 @technosophos @mattfarina hey guys, can we proceed with the pull request please?

**Comment by joejulian** [comment]
I'm going to close and reopen this to see if it'll kick CI.

**Comment by joejulian** [comment]
Just get CI to pass. 🙂

**Comment by mattfarina** [comment]
@mih-kopylov CI is failing due to some lint issues. The fixes should be fairly simple.

**Comment by mih-kopylov** [comment]
@mattfarina I've merged `main` branch, and now it says `This workflow requires approval from a maintainer.`. Could you please approve the CI to run?
https://github.com/helm/helm/actions/runs/5083503093

**Comment by mih-kopylov** [comment]
The linter issues from https://github.com/helm/helm/actions/runs/4920856776/jobs/9131815145 have been fixed.

**Comment by mih-kopylov** [comment]
@mattfarina the issues from https://github.com/helm/helm/actions/runs/5083617601/jobs/9158290502?pr=11631 have been fixed as well :)

**Comment by mih-kopylov** [comment]
@mattfarina @joejulian could you please approve the workflow to run to see if the linting succeeds? And probably approve the pull request? :)

**Comment by joejulian** [comment]
@mih-kopylov This broke main so I had to revert it. Can you rebase (on the revert if it's not merged yet) and try again?

**Comment by mih-kopylov** [comment]
@joejulian here's the new PR #12274 

It's green now, all the checks passed.

**Pull request merged**
